xamarin/Xamarin.Forms
# [iOS] Allow VC renderers of FlyoutPage to specify UIStatusBarStyle

### Description of Change ###

Since we were wrapping the renderers of FlyoutPage and DetailPage on a internal ViewController, the users couldn't provide overrides for `PreferredStatusBarStyle()`.
To fix this we override `ChildViewControllerForStatusBarStyle` on our internal `ChildViewController` and point to page's VC.

### Issues Resolved ### 

- fixes #13053 

### API Changes ###
 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

Should be able to provide UIStatusBar color

### Before/After Screenshots ### 

Before:
<img width="946" alt="Screenshot 2020-12-14 at 23 36 46" src="https://user-images.githubusercontent.com/1235097/102148916-3f53af00-3e65-11eb-9c33-6b292f3dc874.png">


After:
<img width="961" alt="Screenshot 2020-12-14 at 23 34 21" src="https://user-images.githubusercontent.com/1235097/102148729-e71cad00-3e64-11eb-8d01-0b5edc8882ab.png">

### Testing Procedure ###

Start the ControlGallery and check if the UIStatus bar shows white content on the left, and black content on the right.

### PR Checklist ###
